src/main/java/org/adoptopenjdk/jheappo/model/HeapGraph.java

```diff
@@ -0,0 +1,208 @@
+package org.adoptopenjdk.jheappo.model;
+
+import org.adoptopenjdk.jheappo.heapdump.*;
+import org.adoptopenjdk.jheappo.io.HeapProfile;
+import org.adoptopenjdk.jheappo.io.HeapProfileHeader;
+import org.adoptopenjdk.jheappo.io.HeapProfileRecord;
+import org.adoptopenjdk.jheappo.objects.*;
+import org.neo4j.graphdb.*;
+import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.io.fs.FileUtils;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class HeapGraph {
+
+    private static final int BATCH_SIZE = 50000;
+    private final File path;
+    HashMap<Long,UTF8String> stringTable = new HashMap<>();
+    HashMap<Long,ClassObject> clazzTable = new HashMap<>();
+    HashMap<Long,Node> clazzNodes = new HashMap<>();
+    HashMap<Long,Long> clazzNames = new HashMap<>();
+    HashMap<Long,Node> instanceNodes = new HashMap<>();
+    HashMap<Long,InstanceObject> oopTable = new HashMap<>();
+    HashMap<Long,LoadClass> loadClassTable = new HashMap<>();
+    HashSet<Long> rootStickClass = new HashSet<>();
+    HashMap<Long,Long> rootJNIGlobal = new HashMap<>();
+    HashMap<Long,Long> rootJNILocal = new HashMap<>();
+    HashSet<Long> rootMonitorUsed = new HashSet<>();
+    HashMap<Long,RootJavaFrame> rootJavaFrame = new HashMap<>();
+    HashMap<Long,RootThreadObject> rootThreadObject = new HashMap<>();
+    HashMap<Long,PrimitiveArray> primitiveArray = new HashMap<>();
+    HashMap<Long,ObjectArray> objectArray = new HashMap<>();
+
+    public HeapGraph(File path) {
+        this.path = path;
+    }
+
+    public void populateFrom(HeapProfile heapDump) throws IOException {
+        FileUtils.deleteRecursively(path);
+        GraphDatabaseService db = new GraphDatabaseFactory().newEmbeddedDatabase(path);
+        createIndexes(db);
+        int count = 0;
+        Transaction tx = db.beginTx();
+        try {
```

Comment:
This needs a refactor into private methods

Reply:
Also a refactor so try with resources can be used.  Implement the `AutoCloseable` interface